subsites/serializers.py

```diff
@@ -26,39 +26,24 @@ def apply_subsite_changes(data, request, instance):
         )
     # checking users perms based on the subsite_one
     if "perms" in data and isinstance(instance, ResourceBase):
-
         if getattr(settings, "SUBSITE_READ_ONLY", False):
             data["perms"] = ["view_resourcebase"]
             data["download_url"] = None
             data["download_urls"] = None
             return data
 
-        allowed_perms = []
-        _, user = check_user_support(request.user)
-        for user_perm in get_compact_perms_list(
-            instance.get_user_perms(user), 
-            instance.resource_type, 
-            instance.subtype,
-            instance.owner == user
-        ):
-            allowed_perms += [
-                user_perm["name"]
-                for _perm in subsite.allowed_permissions
-                if _perm in user_perm.values()
-            ]
-
-        data["perms"] = list(
-            set(
-                itertools.chain.from_iterable(
-                    filter(None, (
-                            _to_extended_perms(_perm, instance.resource_type)
-                            for _perm in allowed_perms
-                        )
-                    )
-                )
+        subsite_allowed_perms = set(
+            itertools.chain.from_iterable(
+                [
```

Comment:
i would keep the filter None, if for any reason _to_extended_perms return none, is going to fail:

```
>>> import itertools
>>> itertools.chain.from_iterable([None,1,3,4])
<itertools.chain object at 0x000001DBAC72DD88>
>>> list(itertools.chain.from_iterable([None,1,3,4]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```